src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -126,19 +138,24 @@ public void doPolling(StaplerRequest req, StaplerResponse rsp, @QueryParameter S
             throw HttpResponses.notFound();
         }
         try {
-            hudson.model.BuildAuthorizationToken.checkPermission(p, authToken, req, rsp);
+            BuildAuthorizationToken.checkPermission(p, authToken, req, rsp);
         } catch (AccessDeniedException x) {
             LOGGER.log(Level.FINE, "on {0} was denied: {1}", new Object[] {job, x.getMessage()});
             throw x;
         }
+        if (!(p instanceof Queue.Task)) {
+            LOGGER.log(Level.FINE, "{0} is not buildable (not a Task)");
+            throw HttpResponses.error(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, new IOException(job + " is not buildable"));
+        }
         if (!p.isBuildable()) {
```

Comment:
this is to prevent a ClassCastException. #isBuildable only make sense for ... buildable items, but who knows ?